src/web/transactions.rs

```diff
@@ -1,187 +1,214 @@
+                .map(Into::into)
+                .collect(),
+        })
+    }
+}
+
+
+#[derive(Serialize)]
+pub struct CommittedTransactionDTO {
+    #[serde(flatten)]
+    base: TransactionBase,
+}
+
+/// Just as [`CommittedTransactionDTO`], but with rejection reason
+#[derive(Serialize)]
+pub struct RejectedTransactionDTO {
+    #[serde(flatten)]
+    base: TransactionBase,
+    rejection_reason: SerScaleHex<TransactionRejectionReason>,
+}
+
+#[derive(Serialize)]
+pub struct TransactionPayloadDTO {
+    account_id: String,
+    instructions: ExecutableDTO,
+    creation_time: Timestamp,
+    time_to_live_ms:  Option<NonZeroU64>,
+    nonce: Option<NonZeroU32>,
+    metadata: UnlimitedMetadata,
+}
+
+
+
+impl TryFrom<TransactionPayload> for TransactionPayloadDTO {
+    type Error = color_eyre::Report;
+
+    fn try_from(payload: TransactionPayload) -> Result<Self, Self::Error> {
+        Ok(Self {
+            account_id: payload.authority.to_string(),
+            instructions: payload.instructions.into(),
+            creation_time: Timestamp::try_from(payload.creation_time_ms)
+                .wrap_err("Failed to map creation_time")?,
+            time_to_live_ms: payload.time_to_live_ms.try_into()?,
+            nonce: payload.nonce,
+            metadata: payload.metadata,
+        })
+    }
+}
+
+/// Reflection of [`Executable`].
+#[derive(Serialize)]
+#[serde(tag = "t", content = "c")]
+pub enum ExecutableDTO {
+    Instructions(Vec<SerScaleHex< InstructionBox>>),
+    /// WASM binary content is omitted for frontend
+    Wasm,
+}
+
+impl From<Executable> for ExecutableDTO {
+    fn from(value: Executable) -> Self {
+        match value {
+            Executable::Instructions(items) => {
+                Self::Instructions(items.into_iter().map(SerScaleHex).collect())
+            }
+            Executable::Wasm(_) => Self::Wasm,
+        }
+    }
+}
+
+#[get("/{hash}")]
+async fn show(
+    app: web::Data<AppData>,
+    hash: web::Path<HashDeser>,
+) -> Result<web::Json<TransactionDTO>, WebError> {
+    let hash = hash.into_inner().0;
+    let tx = app
+        .iroha_client
+        .request(QueryBuilder::new(FindTransactionByHash::new(HashOf::from_untyped_unchecked(hash)))) // deprecated 
+        .await
+        .map_err(WebError::expect_iroha_find_error)?
+        .only_output();
+
+    Ok(web::Json(
+        tx.try_into().wrap_err("Failed to map TransactionValue")?,
+    ))
+}
+
+#[get("")]
+async fn index(
+    app: web::Data<AppData>,
+    pagination: web::Query<PaginationQueryParams>,
+) -> Result<web::Json<Paginated<Vec<TransactionDTO>>>, WebError> {
+    let Paginated { data, pagination } = app
+        .iroha_client
+        .request(QueryBuilder::new(FindAllTransactions).with_pagination(pagination.0.into()))
+        .await
+        .map_err(WebError::expect_iroha_any_error)?
+        .try_into()?;
+
+    let data = data
+        .into_iter()
+        .map(|x| TryInto::try_into(x)) // transactionQueryResult returned 

```

Comment:
> transactionQueryResult returned

Not really, it should automatically infer that the type is `TransactionDTO`. However, it would be clearer to re-write it more explicitly:
```suggestion
        .map(TransactionDTO::try_from)

```